emData/generate_TP.py

```diff
@@ -65,6 +65,49 @@ class ProjoutIndexDisk(Enum):
     D5PHID = 19
     N_PROJOUT_DISK = 20
 
+#FIXME magic numbers for array sizes:
```

Comment:
The only thing that troubles me in this PR is the large number of hard-wired constants to specify the length of the LUT tables. 
One way to avoid this would be for https://github.com/cms-L1TK/firmware-hls/blob/SWCompileUpdate/emData/generate_TP.py#L271 , to read the .tab file and determine its length, just before line 271 etc. where it needs the hard-wired length.
(Alternatively, maybe Anders or someone knows LUT table length from first principles)

Reply:
added code that reads file lengths from .tab files.

these LUT sizes are pretty hard-coded in many other places in the firmware, but this will hopefully be addressed if we can figure out how to calculate them within the HLS repo.